src/app/domain/icurrentpart.ts

```diff
@@ -3,28 +3,29 @@ import { FirebaseJSONObject, JSONValueWithoutArray } from 'src/app/utils/utils';
 import { Request } from './request';
 import { DomainWrapper } from './DomainWrapper';
 import { Time } from './Time';
+import { MGPOptional } from '../utils/MGPOptional';
 
 export interface IPart extends FirebaseJSONObject {
-    readonly typeGame: NonNullable<string>, // the type of game
-    readonly playerZero: NonNullable<string>, // the id of the first player
-    readonly turn: NonNullable<number>, // -1 means the part has not started, 0 is the initial turn
-    readonly result: NonNullable<IMGPResult>,
+    readonly typeGame: string, // the type of game
+    readonly playerZero: string, // the id of the first player
+    readonly turn: number, // -1 means the part has not started, 0 is the initial turn
+    readonly result: IMGPResult,
     // TODO: should be ReadonlyArray, but does not compile with it!
-    readonly listMoves: NonNullable<Array<NonNullable<JSONValueWithoutArray>>>,
+    readonly listMoves: Array<JSONValueWithoutArray>,
 
     readonly playerOne?: string, // the id of the second player
     /* Server time being handled on server by firestore, when we send it, it's a FieldValue
      * so firebase write the server time and send us back a timestamp in the form of Time
      */
     readonly beginning?: firebase.firestore.FieldValue | Time,
-    readonly lastMoveTime?: firebase.firestore.FieldValue | Time,
+    readonly lastMoveTime?: firebase.firestore.FieldValue | Time | null, // TODO FOR REVIEW: when is it null? null in a firebase json object should only be used to remove a field
```

Comment:
178 is too long, 120 is the maximum.
Le fameux proverbe de la Moria.

Reply:
Oups j'avais pas lu. lastMoveTime est remis à zéro par firebase quand on lui demande de nous fournir ce db-timestamp. Et au push juste après il nous l'envois.
Somme tout soit ça vaut firebase.firestore.FieldValue quand on push une demande de new timestamp,
Et ça vaut qqch au format time ou null quand on reçois la valeur.

Reply:
ça serait pas mieux d'utiliser 0 plutôt que null pour "remettre à zero" ?

Reply:
Bah non car ça ce serait un timestamp, l'enlever et dire "premier janvier 1970" c'est pas pareil.
Aussi, c'est firestore qui a décidé ça.

Reply:
ok TODO enlevé !